src/rpc/api/interfaces/validator.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * The API interface defines the calls that can be made from a Validator
+ */
+import {
+  Attestation, AttestationData,
+  BeaconBlock, bytes, bytes32, bytes48, Fork, IndexedAttestation, number64, Shard, Slot, SyncingStatus, uint64,
+  ValidatorDuty
+} from "../../../types";
+
+export interface IValidatorApi {
+  /**
+   * Requests that the BeaconNode identify information about its implementation in a format similar to a HTTP User-Agent field.
+   * @returns {Promise<bytes32>} An ASCII-encoded hex string which uniquely defines the implementation of the BeaconNode and its current software version.
+   */
+  getClientVersion(): Promise<bytes32>;
+
+  /**
+   * Requests the BeaconNode to provide which fork version it is currently on.
+   * @returns {Promise<{fork: Fork; chainId: uint64}>}
+   */
+  getFork(): Promise<{fork: Fork; chainId: number64}>;
+
+  /**
+   * Requests the genesis_time parameter from the BeaconNode, which should be consistent across all BeaconNodes that follow the same beacon chain.
+   * @returns {Promise<uint64>} The genesis_time, which is a fairly static configuration option for the BeaconNode.
+   */
+  getGenesisTime(): Promise<number64>;
+
+  /**
+   * Requests the BeaconNode to describe if it's currently syncing or not, and if it is, what block it is up to. This is modelled after the Eth1.0 JSON-RPC eth_syncing call.
+   * @returns {Promise<boolean | SyncingStatus>} Either false if the node is not syncing, or a SyncingStatus object if it is.
+   */
+  getSyncingStatus(): Promise<boolean | SyncingStatus>;
+
+  /**
+   * Requests the BeaconNode to provide a set of “duties”, which are actions that should be performed by ValidatorClients. This API call should be polled at every slot, to ensure that any chain reorganisations are catered for, and to ensure that the currently connected BeaconNode is properly synchronised.
+   * @param {bytes48[]} validatorPubkeys
+   * @returns {Promise<{currentVersion: bytes4; validatorDuties: ValidatorDuty[]}>} A list of unique validator public keys, where each item is a 0x encoded hex string.
+   */
+  getDuties(validatorPubkeys: bytes48[]): Promise<{currentVersion: Fork; validatorDuties: ValidatorDuty[]}>;
```

Comment:
Why array of pubkeys, shouldn't validator just get his ValidatorDuty? Probably better to replace bytes46 with BLSPubKey type

Reply:
The idea would be that in the future (we can remove for now) a validator client **could** be responsible for more than 1 validator client imo

Reply:
i.e one function for propose_at_slot and the other for get_committee_assignment?

Reply:
Yeah, thats what I was thinking

Reply:
implemeted